### ISSUE TITLE
'unable to open database file'

### ISSUE THREAD

**Issue by Xtremrules** [codeprefect/multitenancy#1]
The application throws an exception: Application startup exception: Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 14: 'unable to open database file'. while tring to run it.

**Comment by Xtremrules** [comment]
The Exception occurs in `StartupExtension.cs` class

```
foreach (var tenant in allTenants)
{
   var context = dbContextFactory.CreateDbContext(tenant, configuration);
   context.Database.Migrate();
}
```

**Comment by ghost** [comment]
Hello @Xtremrules, did you at all define any tenant(s) in your `Config/tenancy.json`?
From my experience here, it automatically creates the databases if it doesn't exist.

**Comment by Xtremrules** [comment]
Yes @codeprefect, This is the content of the `tenancy.json`

```
[{
        "Id": 1,
        "Name": "Default",
        "Host": "default.tenancy.localhost",
        "ConnectionString": "db\\default.db",
        "Theme": "blue"
    },
    {
        "Id": 2,
        "Name": "Tenant 1",
        "Host": "one.tenancy.localhost",
        "ConnectionString": "db\\tenant1.db",
        "Theme": "green"
    },
    {
        "Id": 3,
        "Name": "Tenant 2",
        "Host": "two.tenancy.localhost",
        "ConnectionString": "db\\tenant2.db",
        "Theme": "red"
    },
    {
        "Id": 4,
        "Name": "Tenant 3",
        "Host": "three.tenancy.localhost",
        "ConnectionString": "db\\tenant3.db",
        "Theme": "red"
    },
    {
        "Id": 5,
        "Name": "Finest Tenant",
        "Host": "four.tenancy.localhost",
        "ConnectionString": "db\\tenant4.db",
        "Theme": "blue"
    }
]
```
It has the same content with the one in the repo.
Is there something am missing?

**Comment by ghost** [comment]
@Xtremrules, can you post the content of your appsettings.json and the exact error you got?

**Comment by ghost** [comment]
@Xtremrules, the issue has been fixed on f039ce2, kindly verify.
Thanks.

**Comment by Xtremrules** [comment]
@codeprefect, The error is resolved, I will study the code tomorrow.

#### Thanks
